lib/Conversion/TorchToMhlo/Linear.cpp

```diff
@@ -102,13 +102,13 @@ RankedTensorType castContractingDim(PatternRewriter &rewriter, Operation *op,
   }
   SmallVector<int64_t> outShape;
   // set batch dims, will skip invalid dimensions
-  for (size_t k = 0; k < lhsShape.size(); ++k) {
+  for (int64_t k = 0; k < static_cast<int64_t>(lhsShape.size()); ++k) {
```

Comment:
it seems weird that you changed the signedness of the iteration variable, and also had to insert a cast as well.